src/commands/src/shared/next-gen-parse.ts

```diff
@@ -1,33 +1,45 @@
+import path from 'path';
+import { Job } from '../../../entities/job';
+import { getDirectory } from '../../../job/jobHandler';
 import { CliCommandResponse, executeCliCommand } from '../helpers';
 
 const RSTSPEC_FLAG = '--rstspec=https://raw.githubusercontent.com/mongodb/snooty-parser/latest/snooty/rstspec.toml';
 interface NextGenParseParams {
-  repoDir: string;
-  commitHash: string;
-  patchId?: string;
+  job: Job;
   isProd?: boolean;
 }
-export async function nextGenParse({
-  repoDir,
-  patchId,
-  commitHash,
-  isProd,
-}: NextGenParseParams): Promise<CliCommandResponse> {
-  const commandArgs = ['build', repoDir, '--output', `${repoDir}/bundle.zip`, RSTSPEC_FLAG];
+export async function nextGenParse({ job, isProd }: NextGenParseParams): Promise<CliCommandResponse> {
+  const repoDir = path.resolve(process.cwd(), `repos/${getDirectory(job)}`);
+  const commitHash = job.payload.newHead;
+  const patchId = job.payload.patch;
```

Comment:
Hey Matt! I just realized that this `job.payload.patch` is the content for the patch file and not the patch ID. The patch ID from the makefile is retrieved [here](https://github.com/mongodb/docs-worker-pool/blob/a720d98b7e58418a92c06f7b4d346e3feacd7fd8/makefiles/shared.mk#L12). It's a bit odd, but we also get the commit hash [here](https://github.com/mongodb/docs-worker-pool/blob/a720d98b7e58418a92c06f7b4d346e3feacd7fd8/makefiles/shared.mk#L1) as well, but I think using the `job.payload.newHead` value is fine since it should be the same as what we get in the makefile. 

Since we aren't using the makefiles, the patch ID is being retrieved from the[prepareBuildAndGetDependencies function](https://github.com/mongodb/docs-worker-pool/blob/d1367a7702a97d2ed26b3e0d1ede9ac7c1cb2248/src/commands/src/helpers/dependency-helpers.ts#L96]. 

Not sure how much this all really matters since I'm assuming only a small number of builds use the patch ID (I'm assuming.

@rayangler @casthewiz, thoughts on this? In what situations do we use a patch? And is it something we may be able to forgo entirely?

Reply:
Iirc, patches were used primarily for pushless staging builds. Pushless staging isn't really used by a lot of people, and seems like it hasn't been regularly used since August 2023. I think we've discussed fully removing support for pushless staging in the future, but not sure when we'll commit to that.